docs/verifiable-document/document-template.md

```diff
@@ -4,8 +4,228 @@ title: Deploying Document Renderer
 sidebar_label: Deploying Document Renderer
 ---
 
-Overview
+OA documents are both readable by machines as well as by humans. Every OA document file is stored in `.json` format, allowing any applications to process the content within. To present the data file in a human readable format, a renderer needs to be written.
 
-## Header
+In this guide, we will build and deploy the renderer to display data from a "Certificate of Completion".
 
-TBD
+This renderer is a static website that will be embedded in compliant OA viewer. It will take in the OA document in the unwrapped form and generates the corresponding HTML code for rendering.
+
+## Pre-requisite
+
+- [git](https://git-scm.com/downloads)
+- [node.js](https://nodejs.org/en/download/)
+- [Github account](https://github.com/)
+- [Netlify account](https://www.netlify.com/)
+- [Basic react.js knowledge](https://reactjs.org/)
+
+## Clone Decentralized Renderer React Template
+
+A template for building your own document renderer has been created for you at https://github.com/Open-Attestation/decentralized-renderer-react-template.
+
+### Clone code repository locally
+
+```sh
+git clone https://github.com/Open-Attestation/decentralized-renderer-react-template.git
+rm -rf .git
+```
+
+### Install code dependencies
+
+```sh
+cd decentralized-renderer-react-template
+npm install
+```
+
+### Run development preview
+
+```sh
+npm run storybook
+```
+
+### Development environment
+
+![Default Story Book View](/docs/verifiable-document/document-template/default-storybook.png)
+
+After running the storybook, you should be able to see the default template provided at http://localhost:6006/.
+
+This is a live preview where you can see the changes when you:
+
+1. edit the unwrapped document data in the "Knobs" tab
+1. edit the template code to render the data
+
+## Developing the renderer
+
+Now that we have set up the development environment, we can start writing our renderer. We will first define the data structure of our Certificate of Completion, followed by writing the renderer to render the HTML code corresponding to the data provided.
+
+### Update sample document data and type
+
+To update the unwrapped document data and the corresponding data type, you will need to update the data definition file in `src/templates/customTemplate/samples.ts`:
+
+```js
+import { Document } from "@govtechsg/decentralized-renderer-react-components";
+
+export interface CustomTemplateCertificate extends Document {
+  name: string;
+  recipient: {
+    name: string
+  };
+}
+
+export const customTemplateCertificate: CustomTemplateCertificate = {
+  name: "OpenAttestation Tutorial Certificate of Completion",
+  recipient: {
+    name: "John Doe"
+  },
+  $template: {
+    name: "COC",
+    type: "EMBEDDED_RENDERER",
+    url: "http://localhost:3000"
+  }
+};
+```
+
+In the above document data, you see three root objects:
+
+### 1. \$template
```

Comment:
I dunno if you missed that question in my other comment, Adding back in case

On a side note, name and recipient are not mandatory fields for oa v2, Not sure why we make it looks so ?

Reply:
Acutally name is really good to have, that is used in the mobile to determine the file name... else it looks like blank. Would it be a problem to make it sound mandatory here? 

For recipient, they really dont need it, if you still read it as mandatory maybe you could suggest alternative wording for that?

Reply:
No recipient is fine :+1: 

For name I'm not sure, not a big deal to me, let's keep it, and see if it causes confusion to anyone